internal/client.go

```diff
@@ -816,7 +821,13 @@ func NewServiceClient(workflowServiceClient workflowservice.WorkflowServiceClien
 	}
 
 	// Create outbound interceptor by wrapping backwards through chain
-	client.interceptor = &workflowClientInterceptor{client: client}
+	client.rootInterceptor = &workflowClientInterceptor{
+		client: client,
+		eagerDispatcher: &eagerWorkflowDispatcher{
+			workersByTaskQueue: make(map[string][]eagerWorker),
+		},
```

Comment:
I think this makes sense on the client itself instead of the client interceptor

Reply:
I preferred it in the interceptor because that is where it is used

Reply:
The interceptor often references the object its intercepting (in this case a client)